packages/account/src/Components/address-details/address-details.tsx

```diff
@@ -7,176 +16,183 @@ import {
-    makeCancellablePromise,
-    PlatformContext,
-    TLocationList,
-} from '@deriv/shared';
+import { getLocation, makeCancellablePromise, PlatformContext } from '@deriv/shared';
 import { splitValidationResultTypes } from '../real-account-signup/helpers/utils';
 import classNames from 'classnames';
+import { observer, useStore } from '@deriv/stores';
+import { FormInputField } from '../forms/form-fields';
+
+export type TAddressDetailFormProps = {
+    address_line_1: string;
+    address_line_2?: string;
+    address_city: string;
+    address_state?: string;
+    address_postcode?: string;
+};
 
 type TAddressDetails = {
-    states_list: TLocationList[];
     getCurrentStep?: () => number;
-    onSave: (current_step: number, values: FormikValues) => void;
+    onSave: (current_step: number, values: TAddressDetailFormProps) => void;
     onCancel: (current_step: number, goToPreviousStep: () => void) => void;
     goToNextStep: () => void;
     goToPreviousStep: () => void;
-    validate: (values: FormikValues) => FormikValues;
+    validate: (values: TAddressDetailFormProps) => TAddressDetailFormProps;
     onSubmit: (
         current_step: number | null,
-        values: FormikValues,
+        values: TAddressDetailFormProps,
         action: (isSubmitting: boolean) => void,
         next_step: () => void
     ) => void;
     is_svg: boolean;
     is_mf?: boolean;
     is_gb_residence: boolean | string;
     onSubmitEnabledChange: (is_submit_disabled: boolean) => void;
-    selected_step_ref?: React.RefObject<FormikProps<FormikValues>>;
+    selected_step_ref?: React.RefObject<FormikProps<TAddressDetailFormProps>>;
     fetchStatesList: () => Promise<unknown>;
-    value: FormikValues;
-};
-
-type TFormValidation = {
-    warnings: { [key: string]: string };
-    errors: { [key: string]: string };
-};
-
-type TInputField = {
-    name: string;
-    required?: boolean | string;
-    label: string;
-    maxLength?: number | string;
-    placeholder: string;
-    onChange?: (e: any) => void;
-    disabled?: string;
+    value: TAddressDetailFormProps;
+    disabled_items: string[];
+    has_real_account: boolean;
 };
 
 type TAutoComplete = {
     value: boolean;
     text: string;
 };
 
-const InputField = (props: TInputField) => {
-    return (
-        <Field name={props.name}>
-            {({ field, form: { errors, touched } }: FormikValues) => (
-                <React.Fragment>
-                    <Input
-                        type='text'
-                        autoComplete='off'
-                        maxLength={props.maxLength || 30}
-                        error={touched[field.name] && errors[field.name]}
-                        {...field}
-                        {...props}
-                    />
-                </React.Fragment>
-            )}
-        </Field>
-    );
-};
+/**
+ * Component to display address details form
+ * @name AddressDetails
+ * @param {Function} getCurrentStep - function to get current step
```

Comment:
Mostly all names of props are self descriptive, why do we need to write so many additional lines of code.
I really don’t understand it, I'm sorry :)

Reply:
@likhith-deriv we can reduce the docs by documenting the type itself as we've typescript as well